src/auth/oauth/oauth.ts

```diff
@@ -239,13 +244,19 @@ const ShopifyOAuth = {
         }
 
         const jwtPayload = decodeSessionToken(matches[1]);
-        currentSessionId = this.getJwtSessionId(jwtPayload.dest.replace(/^https:\/\//, ''), jwtPayload.sub);
+        const shop = jwtPayload.dest.replace(/^https:\/\//, '');
+        if (isOnline) {
+          currentSessionId = this.getJwtSessionId(shop, jwtPayload.sub);
+        } else {
+          currentSessionId = this.getOfflineSessionId(shop);
+        }
       }
     }
 
     // We fall back to the cookie session to allow apps to load their skeleton page after OAuth, so they can set up App
     // Bridge and get a new JWT.
```

Comment:
This comment isn't quite correct - it's not just the post-OAuth, pre-JWT embedded scenario, it's also the non-embedded scenario, right?

Reply:
True - the comment applies to embedded apps when it falls back, but it always applies for non-embedded ones. Will reword it (hopefully we can remove it soon).